puppet/lib/puppet/reports/puppetdb.rb

```diff
@@ -40,21 +40,28 @@ def report_to_hash
       resources = build_resources_list
       is_noop = resources.any? { |rs| has_noop_event?(rs) } and resources.none? { |rs| has_failed_event?(rs) }
 
+
+      defaulted_catalog_uuid = defined?(catalog_uuid) ? catalog_uuid : transaction_uuid
```

Comment:
Hm I think that would require us to refactor report handling in the terminus quite a bit, right now on catalogs we call `to_data_hash` and handle the catalog like we would any ordinary hash. For reports we use the instance variables on the Puppet class to access the fields. If we wanted to do that, we should probably do that in another ticket.